code/addons/backgrounds/src/decorators/withGrid.ts

```diff
@@ -18,18 +18,42 @@ export const withGrid = (StoryFn: StoryFunction<Renderer>, context: StoryContext
   const offsetY = gridParameters.offsetY ?? (isInDocs ? 20 : defaultOffset);
 
   const gridStyles = useMemo(() => {
-    const selector =
-      context.viewMode === 'docs' ? `#anchor--${context.id} .docs-story` : '.sb-show-main';
-
     const backgroundSize = [
       `${cellSize * cellAmount}px ${cellSize * cellAmount}px`,
       `${cellSize * cellAmount}px ${cellSize * cellAmount}px`,
       `${cellSize}px ${cellSize}px`,
       `${cellSize}px ${cellSize}px`,
     ].join(', ');
 
+    const parentSelector =
+      context.viewMode === 'docs' ? `#anchor--${context.id} .docs-story` : '.sb-show-main';
+
     return `
-      ${selector} {
+      /** this ensures the we fill the whole space **/
+      html {
+        height: 100%;
+      }
+
+      ${parentSelector} {
+        position: relative;
+        height: 100%;
+      }
```

Comment:
I'm not so sure about this.

I suspect this might cause unexpected behavior.

Reply:
Without this, would the grid break when used on docs pages?

Reply:
yes. without this, the grid would be on top of all the other elements so there's no way to interact to elements within it, if any.